### ISSUE TITLE
version changing workflow and version inheritance

### ISSUE THREAD

**Issue by nazrhyn** [apidoc/apidoc#205]
As I'm building out the documentation for my team's API, I've had some trouble visualizing what the version changing workflow would be. This is an example of what it appears to be:

1. Write initial documentation.    
   All blocks that should be versioned have `@apiVersion 1.0.0`.    
   The global `version` is set to `1.0.0`.
1. API version changes to **1.1.0**.
1. For all API routes that have changed:
   1. Copy the old headers to the "old" bucket.
   1. Update the live headers to incorporate the **1.1.0** changes.
1. Update the global `version` setting to `1.1.0`.
1. Update *every* live header to `@apiVersion 1.1.0`.

The last step in that that process is going to be somewhat arduous. There will be a massive amount of headers that will need to be individually updated every time we increase our API version.

With regard to the `version` setting: currently, it looks like all that setting does is help to populate the master version button at the top of the template. All header blocks that *do not* have a defined version default to **0.0.0**.

Would it be easier/nicer for blocks that do not have a defined version to *inherit* the value of the `version` setting? In that case, `@apiVersion` would only be added to blocks that have ever changed and needed to be versioned. All other blocks would simply inherit the global version and have no change history. The workflow would change to:

1. Write initial documentation.    
   No blocks have `@apiVersion`.    
   The global `version` is set to `1.0.0`.
1. API version changes to **1.1.0**.
1. For all API routes that have changed:
   1. Add `@apiVersion 1.0.0` to the header and copy it to the "old" bucket.
   1. Update the live headers to incorporate the **1.1.0** changes and to change to `@apiVersion 1.1.0`.
1. Update the global `version` setting to `1.1.0`.

The difference between the two is that, in one case, every time the API version changes, the `@apiVersion` of every single header has to be updated; whereas, in the other case, only those headers that have had documentation changes need to be touched. All of the other headers will follow the "global" version.

What do you think?

**Comment by rottmann** [comment]
That makes sense.

The current version (thats why i create it) was only for documenting internal changes for an internal dev team, but this will still work with your change of version inheritance.

I think about to add a @apiSince or @apiSinceVersion too, so the workflow could be simplified.

**Comment by nazrhyn** [comment]
Great.

Yeah, we're going to be using this for customer-facing documentation of our API. Anything that allows the whole "update" workflow to simplify will be great.

Thanks for your work on this; it's a nice system.

**Comment by nazrhyn** [comment]
Does your comment on #212 invalidate this issue as well?

**Comment by rottmann** [comment]
Yes and no, the @apiSince idea is invaldiate when somebody use the apiDoc versioning feature.
But without the versioning feature the @apiSince make sense.

**Comment by nazrhyn** [comment]
Anything that simplifies the process of upgrading versions. We've got 68 API routes documented and still have a good bit left to go; that's a lot of find/replace and possible mistakes.

Thanks for continuing to consider this!

**Comment by cladera** [comment]
Hi there! Was @apiSince finally implemented?
